nuklear_gamepad.h

```diff
@@ -72,6 +72,21 @@ enum nk_gamepad_button {
     NK_GAMEPAD_BUTTON_LAST
 };
 
+struct nk_gamepads;
+
+typedef nk_bool (*nk_gamepad_init_fn)(void* user_data, struct nk_gamepads *gamepads);
+typedef void (*nk_gamepad_update_fn)(void* user_data, struct nk_gamepads *gamepads);
+typedef void (*nk_gamepad_free_fn)(void* user_data, struct nk_gamepads *gamepads);
+typedef const char *(*nk_gamepad_name_fn)(void* user_data, struct nk_gamepads *gamepads, int num);
```

Comment:
Thanks so much, this is great! I like the idea of being able to change the callbacks so that you could flip input drivers as needed.

Could we send the `user_data` parameter to the last parameter? So it's...
```c
typedef nk_bool (*nk_gamepad_init_fn)(struct nk_gamepads *gamepads, void* user_data);
```

Reply:
Sure, done